membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -49,71 +48,89 @@ class AttributeController extends Controller with LoggingWithLogstashFields {
-
-      def withTimer[R](whichCall: String, result: Future[Disjunction[String, R]]) = {
-        import loghandling.StopWatch
-        val stopWatch = new StopWatch
-
-        result.map { res: Disjunction[String, R] =>
-          val latency = stopWatch.elapsed
-          val customFields: List[LogField] = List("zuora_latency_millis" -> latency.toInt, "zuora_call" -> whichCall, "identityId" -> identityId)
-          res match {
-            case -\/(a) => logErrorWithCustomFields(s"$whichCall failed. Msg: ${a}", customFields)
-            case \/-(_) => logInfoWithCustomFields(s"$whichCall took ${latency}ms", customFields)
+    backendAction.async { implicit request =>
+      authenticationService.userId(request) match {
+        case Some(identityId) =>
+          pickAttributes(identityId, doExperiment).map {
+            case Some(attrs @ Attributes(_, Some(tier), _, _, _, _, _)) =>
+              log.info(s"$identityId is a member - $endpointDescription - $attrs")
+              onSuccessMember(attrs).withHeaders(
+                "X-Gu-Membership-Tier" -> tier,
+                "X-Gu-Membership-Is-Paid-Tier" -> attrs.isPaidTier.toString
+              )
+            case Some(attrs) =>
+              log.info(s"$identityId is a contributor - $endpointDescription - $attrs")
+              onSuccessMemberAndOrContributor(attrs)
+            case _ =>
+              onNotFound
           }
+        case None =>
+          metrics.put(s"$endpointDescription-cookie-auth-failed", 1)
+          Future(unauthorized)
+
         }
-        result
       }
+  }
 
 
-      def queryToAccountIds(response: QueryResponse): List[AccountId] =  response.records.map(_.Id)
+  private def attributesFromZuora(identityId: String, zuoraRestService: ZuoraRestService[Future], subscriptionService: SubscriptionService[Future]): Future[Option[Attributes]] = {
 
-      def getSubscriptions(accountIds: List[AccountId]): Future[Disjunction[String, List[Subscription[AnyPlan]]]] = {
-        def sub(accountId: AccountId): Future[List[Subscription[AnyPlan]]] = {
-          subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
-        }
-        val maybeSubs: Future[List[Subscription[AnyPlan]]] = Future.traverse(accountIds)(id => sub(id)).map(_.flatten)
+    def withTimer[R](whichCall: String, result: Future[Disjunction[String, R]]) = {
+      import loghandling.StopWatch
+      val stopWatch = new StopWatch
 
-        maybeSubs map { subs =>
-          if (subs.isEmpty) \/.left(s"Error getting subscriptions for identityId $identityId") else \/.right(subs)
+      result.map { res: Disjunction[String, R] =>
+        val latency = stopWatch.elapsed
+        val customFields: List[LogField] = List("zuora_latency_millis" -> latency.toInt, "zuora_call" -> whichCall, "identityId" -> identityId)
+        res match {
+          case -\/(a) => logErrorWithCustomFields(s"$whichCall failed. Msg: ${a}", customFields)
```

Comment:
look good with the logging, just be careful of logging error level as we can log PII to sentry by mistake (warn level only goes to kibana)